ciphey/basemods/Decoders/brainfuck.py

```diff
@@ -0,0 +1,172 @@
+from typing import Optional, Dict, List, Tuple
+
+from ciphey.iface import Config, ParamSpec, T, U, Decoder, registry, WordList
+from ciphey.common import fix_case
+import time
+
+
+@registry.register
+class Brainfuck(Decoder[str, str]):
+    def decode(self, ctext: T) -> Optional[U]:
+        """
+        Takes a ciphertext and treats it as a Brainfuck program,
+        interpreting it and saving the output as a string to return.
+
+        Brainfuck is a very simple, Turing-complete esoteric language.
+        Below is a simplified interpreter that attempts to check whether a
+        given ciphertext is a brainfuck program that would output a string.
+
+        A program that can be "decoded" like this is one that:
+            * Does not require user input ("," instruction)
+            * Includes at least one putchar instruction (".")
+            * Does not contain anything but the main 7 instructions,
+                (excluding ",") and whitespace (tabs, spaces, and newlines)
+
+        Details:
+            * This implementation wraps the memory pointer for ">" and "<"
+            * It is time-limited to 60 seconds, to prevent hangups
+            * The program starts with 100 memory cells, chosen arbitrarily
+        """
+
+        result = ""
+        memory = [0] * 100
+        codeptr, memptr = 0, 0      # instruction pointer, stack pointer
+        timelimit = 60              # timeout, in seconds
+
+        bracemap, isbf = self.bracemap_and_check(ctext)
+
+        # this doesn't appear to be a brainfuck program we can decode
+        if not isbf:
+            return None
+
+        # get start time
+        start = time.time()
+
+        while codeptr < len(ctext):
+
+            current = time.time()
+
+            # arbitrarily quit if we've been running for over a minute
+            if current - start > timelimit:
+                return None
```

Comment:
Perhaps add some logging here so it says if it failed because it ran for too long when using -vv.

Reply:
Definitely a good call.